.spin/cmds.py

```diff
@@ -567,3 +568,67 @@ def _config_openblas(blas_variant):
         os.makedirs(openblas_dir, exist_ok=True)
         with open(pkg_config_fname, "wt", encoding="utf8") as fid:
             fid.write(openblas.get_pkg_config().replace("\\", "/"))
+
+
+@click.command()
+@click.option(
+    "-v", "--version-override",
+    help="NumPy version of release",
+    required=False
+)
+@click.pass_context
+def notes(ctx, version_override):
+    """🎉 Generate release notes and validate
+
+    \b
+    Example:
+
+    \b
+    $ spin notes --version-override 2.0
+
+    \b
+    To automatically pick the version
+
+    \b
+    $ spin notes
+    """
+    if not version_override:
+        version = util.get_config()['project']['version']
+    else:
+        version = version_override
+
+    click.secho(
+        f"Generating release notes for NumPy {version}",
+        bold=True, fg="bright_green",
+    )
+
+    # Check if `towncrier` is installed
+    if not shutil.which("towncrier"):
+        raise click.ClickException(
+            f"please install `towncrier` to use this command"
+        )
+
+    # towncrier build --version 2.1 --yes
+    cmd = ["towncrier", "build", "--version", version, "--yes"]
+    try:
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8")
```

Comment:
Consider using `_run` as elsewhere in the file.

Reply:
Adde via [123af65](https://github.com/numpy/numpy/pull/25017/commits/123af65fb1cb6b7695b58f74adf3645cbfb42f2b)